### PR TITLE
Feat/dedicated exchange

### PR DESCRIPTION
**Pull request by FredericEspiau**
### Description of the Change

Also tiny rewriting of `products.service.js`.

### CODE REVIEW

File: client/app/components/sidebar-menu/sidebar-menu.config.js [hunk 1]
```diff
@@ -250,12 +250,18 @@ angular.module("App").run(($q, SidebarMenu, Products, User, constants, translato
             loadOnState: "app.microsoft.exchange"
         }, microsoftItem);
 
+        const exchangeProductTypes = {
```
Comment (FredericEspiau) [inline client/app/components/sidebar-menu/sidebar-menu.config.js]: I've kept the "default state" feature, so if a new offer arrives, then the `Hosted` state is selected by default

File: client/app/components/product/product.service.js [hunk 2]
```diff
@@ -174,44 +174,54 @@ angular.module("services").service("Products", [
...
-                    selectedProduct.type = $rootScope.currentSectionInformation ? $rootScope.currentSectionInformation.toUpperCase() : null;
-                    selectedProduct.organization = $stateParams.organization ? $stateParams.organization : "";
-
-                    if (selectedProduct.name === "") {
-                        return {
-                            name: "",
-                            organization: "",
-                            type: ""
-                        };
+            return this.getProducts(forceRefresh)
```
Comment (FredericEspiau) [inline client/app/components/product/product.service.js]: I've rewritten it as I couldn't figure out what it was supposed to do

File: bower.json [hunk 3]
```diff
@@ -113,8 +113,9 @@
     "angular-ui-router": "^1.0.0",
     "animate.css": "~3.5.0",
     "ng-at-internet": "~3.0.0",
-    "ovh-ui-angular": "^2.7.x",
-    "ovh-ui-kit": "^2.7.x"
+    "ovh-api-services": "^2.1.9",
```
Comment (antleblanc) [inline bower.json]: ⚠️ `ovh-api-services` has been upgraded to `v3.0.0` (ref: #377).

**Comment by FredericEspiau** [comment]
QC is OK but I'm leaving the "Quality Check NOK" tag as the API doesn't work yet @jleveugle

**Comment by FredericEspiau** [comment]
I've switched the Quality Check tag to OK as it can work without the API, can anyone please prod this ? 😃 

@jleveugle @Netnix

File: client/app/components/product/product.service.js [hunk 4]
```diff
@@ -174,44 +174,54 @@ angular.module("services").service("Products", [
...
-                        let found;
-
-                        while (!found && i < productsList[productLength].subProducts.length) {
-                            if (productsList[productLength].subProducts[i].name === selectedProduct.name && productsList[productLength].subProducts[i].type === selectedProduct.type) {
-                                found = productsList[productLength].subProducts[i];
-                            }
-                            i++;
+                    let productMatchingSelectedProduct = null;
+
+                    _(productsList).forEach((product) => {
```
Comment (FredericEspiau) [inline client/app/components/product/product.service.js]: Might be a way to do this recursively but I couldn't figure it out

File: client/app/components/user/session.service.js [hunk 5]
```diff
@@ -59,12 +59,18 @@ class SessionService {
     }
 
     getMicrosoftMenu (products) {
+        const exchangeProductTypes = {
```
Comment (FredericEspiau) [inline client/app/components/user/session.service.js]: Same logic in two different files... There is place for improvement there

**Pull request merged**